### PR TITLE
NIFI-9514 NIFI-9515: Add UI support for Parameter Providers in Controller Services

### PR DESCRIPTION
**Pull request by mtien-apache**
Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

This PR adds a new Parameter Providers tab in the Controller Settings UI dialog. From the Parameter Providers tab, users can add and configure Parameter Providers to fetch external parameters. Parameter contexts created from a provider will be listed in Parameter Contexts, where the parameters will be in a read-only state.

~~- To test, you'll need to check out and rebase on top of [PR 5369](https://github.com/apache/nifi/pull/5369).~~ This PR is rebased on `main`, which now includes PR 5369.
- Links to Jiras: [NIFI-9514](https://issues.apache.org/jira/browse/NIFI-9514), [NIFI-9515](https://issues.apache.org/jira/browse/NIFI-9515)

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [x] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by mcgilman** [comment]
Reviewing...

**Comment by mtien-apache** [comment]
Please do not merge yet. This is still a work in progress.

**Comment by mtien-apache** [comment]
This PR is now ready for review.

**Review by mcgilman — CHANGES_REQUESTED** [review]
Thanks @mtien-apache! Great first pass here. The capability generally works well but I noticed a couple of things that we want to address before merging. I commented directly on a couple of parts of the code below. Beyond these we need to better handle when a managed authorized is used. Specifically...

- We need to support managing the permission of each Parameter Provider.
- In the View/Edit Parameter Provider dialogs, we need to handle when the current user does not have permissions to any referenced Parameter Contexts. This shouldn't prevent changing the configuration for the Parameter Provider.
- The Fetch dialog should only be available for users that have READ for the Parameter Provider and READ/WRITE of any mapped Parameter Contexts. We'll also need to ensure they have permission to create a Parameter Contexts.
- Once the Fetch dialog is open, the back end should be able to provide guidance about whether the Apply button should be active. This will be based on what Parameter values are changing. Additionally, if the user has opted to change the sensitive flag they would need permissions to all components that reference that Parameter.

I also noticed a case where the table scrolls when it shouldn't.

scrolling-table

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-settings.js
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-settings.js:2585]: The user guide is still referencing `Reporting Task Controller Services`.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/jquery/statusbar/jquery.statusbar.js
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/jquery/statusbar/jquery.statusbar.js:36]: I suspect the references to `nf.CanvasUtils` here are a copy/paste issue. Generally, our `jquery` plugins do not reference any `nf` classes. The intended direction is for our `nf` classes to reference and use the `jquery` plugins. Is it possible here to instead of calling the formatting code here to instead let the client of the `statusbar` format the bulletins and pass in a `jquery` reference (`$`)?

**Review by mcgilman — COMMENTED** [review]
Thanks for the update @mtien-apache! I've left a few more comments below. I've also hit the issue addressed in this PR [1]. Once this PR is in, I'll try running against #6388.

[1] https://github.com/apache/nifi/pull/6388

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-parameter-contexts.js
Comment (mtien-apache) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-parameter-contexts.js]: Fixed.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-parameter-contexts.js
Comment (mtien-apache) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-parameter-contexts.js]: Fixed.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-settings.js
Comment (mtien-apache) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-settings.js:2573]: This will be addressed in a separate Jira, which will include new screenshots too.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/jquery/statusbar/jquery.statusbar.js
Comment (mtien-apache) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/jquery/statusbar/jquery.statusbar.js]: Your observation is correct. It was a copy/paste issue. I fixed this by passing in formatted bulletins and reverted the jquery reference to its original method.

**Comment by mtien-apache** [comment]
@mcgilman Thanks for reviewing! I addressed all of your comments. 

> I've also hit the issue addressed in this PR [1]. Once this PR is in, I'll try running against https://github.com/apache/nifi/pull/6388.

For this issue, I also tested my PR against PR 6388 and it works as expected. 

Will you review again?

**Review by mcgilman — APPROVED** [review]
Thanks @mtien-apache! Nice work!

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-settings.js
Comment (mtien-apache) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-settings.js]: Here is the Jira: https://issues.apache.org/jira/browse/NIFI-10501

**Pull request merged**
